src/Nix/Expr/Types.hs

```diff
@@ -201,7 +201,17 @@ instance IsString (Params r) where
 
 -- | 'Antiquoted' represents an expression that is either
 -- antiquoted (surrounded by ${...}) or plain (not antiquoted).
-data Antiquoted (v :: *) (r :: *) = Plain !v | EscapedNewline | Antiquoted !r
+data Antiquoted (v :: *) (r :: *)
+  = Plain !v
+  | EscapedNewline
+  -- ^ 'EscapedNewline' corresponds to the special newline form
+  --
+  -- > ''\n
+  --
+  -- in an indented string. It is equivalent to a single newline character:
+  --
+  -- > ''''\n'' === "\n"
```

Comment:
I used `git blame` to figure this out: https://github.com/haskell-nix/hnix/commit/01d1ae99a79d1623ee94befc0a29d2b4e7d542e2 ;)